Spigot-Server-Patches/0410-Add-root-admin-user-detection.patch

```diff
@@ -0,0 +1,84 @@
+From 9ede8411afc9a5f84436f9aea86e8447ea24d72e Mon Sep 17 00:00:00 2001
+From: egg82 <eggys82@gmail.com>
+Date: Thu, 8 Aug 2019 14:12:48 -0600
+Subject: [PATCH] Add root/admin user detection
+
+This patch detects whether or not the server is currently executing as a privileged user and spits out a warning.
+The warning serves as a sort-of PSA for newer server admins who don't understand the risks of running as root.
+We've seen plenty of bad/malicious plugins hit markets, and there's been a few close-calls with exploits in the past.
+Hopefully this helps mitigate some potential damage to servers, even if it is just a warning.
+
+diff --git a/src/main/java/com/destroystokyo/paper/util/ServerEnvironment.java b/src/main/java/com/destroystokyo/paper/util/ServerEnvironment.java
+new file mode 100644
+index 000000000..6364ea403
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/util/ServerEnvironment.java
+@@ -0,0 +1,36 @@
++package com.destroystokyo.paper.util;
++
++import java.io.OutputStream;
++import java.io.PrintStream;
++import java.util.prefs.Preferences;
++
++public class ServerEnvironment {
++    private static final boolean runningAsRootOrAdmin;
++
++    static {
++        // https://stackoverflow.com/a/23538961
++        Preferences prefs = Preferences.systemRoot();
++        PrintStream err = System.err;
++        synchronized (err) { // Synchronized to prevent swapping back to "err" before printing the warning from the try/catch to console
```

Comment:
for the synchronized block to be effective, **if it even is effective at all**, you must re-check the current value of System.err while in it

Reply:
Did some research and testing, looks like a flush() will work fine.